services/workflows-service/src/data-analytics/data-analytics.service.ts

```diff
@@ -1,11 +1,148 @@
 import { Injectable } from '@nestjs/common';
-import { DataAnalyticsRepository } from './data-analytics.repository';
 import { PrismaService } from '@/prisma/prisma.service';
+import { InlineRule, TransactionsAgainstDynamicRulesType } from './types';
+import { AggregateType } from './consts';
+import { Prisma } from '@prisma/client';
 
 @Injectable()
 export class DataAnalyticsService {
-  constructor(
-    protected readonly prisma: PrismaService,
-    private dataAnalyticsRepository: DataAnalyticsRepository,
-  ) {}
+  private _evaluateNameToFunction: Record<string, Function> = {};
+
+  constructor(protected readonly prisma: PrismaService) {
+    this._evaluateNameToFunction[this.evaluateTransactionsAgainstDynamicRules.name] =
+      this.evaluateTransactionsAgainstDynamicRules.bind(this);
+  }
+
+  async runInlineRule(projectId: string, inlineRule: InlineRule) {
+    const evaluateFn = this._evaluateNameToFunction[inlineRule.fnName];
+    if (!evaluateFn) {
```

Comment:
In general please make sure to separate variables from statements like `if` or loops with a linebreak. 🙏🏼 
```suggestion
    const evaluateFn = this._evaluateNameToFunction[inlineRule.fnName];
    
    if (!evaluateFn) {
```